src/sabre_swap/mod.rs

```diff
@@ -162,7 +154,7 @@ pub fn build_swap_map(
     layout: &mut NLayout,
     num_trials: usize,
 ) -> (SwapMap, PyObject) {
-    let run_in_parallel = getenv_use_multiple_threads();
+    let run_in_parallel = getenv_use_multiple_threads() && num_trials > 1;
```

Comment:
Ah that's a good catch, I guess we were potentially creating an unnecessary thread pool before for the single trial case.